### PR TITLE
anagram: implement exercise

### PR DESCRIPTION
**Pull request by dreig**

### CODE REVIEW

**Comment by rlmark** [comment]
This one I can't re-open because the branch was "force pushed or recreated" according to github. 🤔 Digging into this a bit, I found some steps [here](https://gist.github.com/robertpainsi/2c42c15f1ce6dab03a0675348edd4e2c) to try.

**Comment by dreig** [comment]
I've followed the first 3 steps from that gist/guide, and force pushed the commit with which I opened this PR in the first place. I think it should be possible to re-open this now.

File: config.json [hunk 1]
```diff
@@ -376,6 +376,14 @@
...
         "prerequisites": [],
         "difficulty": 1
+      },
+      {
+        "slug": "anagram",
+        "name": "Anagram",
+        "uuid": "d98be569-4068-4367-b04a-55ecbf438971",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 1
```
Comment (SimaDovakin) [inline config.json:386]: More appropriate difficulty is 2 or 3 I guess. Could you set the difficulty to 3, please?

File: exercises/practice/anagram/anagram.test.u [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  actual = findAnagrams "go" ["go Go GO"]
+  expected = []
+  Test.label "does not detect an anagram if the original word is repeated" <| Test.expect (actual === expected)
+
+anagram.test.ex13 = let
+  actual = findAnagrams "tapper" ["patter"]
+  expected = []
+  Test.label "anagrams must use all letters exactly once" <| Test.expect (actual === expected)
+
+anagram.test.ex14 = let
```
Comment (SimaDovakin) [inline exercises/practice/anagram/anagram.test.u:65]: This test case should be removed too.

File: exercises/practice/anagram/anagram.test.u [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  actual = findAnagrams "BANANA" ["Banana"]
+  expected = []
+  Test.label "words are not anagrams of themselves even if letter case is partially different" <| Test.expect (actual === expected)
+
+anagram.test.ex17 = let
+  actual = findAnagrams "BANANA" ["banana"]
+  expected = []
+  Test.label "words are not anagrams of themselves even if letter case is completely different" <| Test.expect (actual === expected)
+
+anagram.test.ex18 = let
```
Comment (SimaDovakin) [inline exercises/practice/anagram/anagram.test.u:85]: And this should also be removed.

**Review by SimaDovakin — CHANGES_REQUESTED** [review]

**Comment by SimaDovakin** [comment]
@dreig In addition, could you resolve conflicts in `config.json`? If all tests will pass I'll merge this PR.

**Comment by dreig** [comment]
sorry, I haven't touched unison since #12in23, and I don't have the time to get back into it and update the PR anytime soon.
Feel free to close this PR and use the source code, if it's in any way useful. I don't care for attribution.

**Comment by SimaDovakin** [comment]
@dreig Ok. Thanks.

**Comment by SimaDovakin** [comment]
I transferred this to https://github.com/exercism/unison/pull/138.

**Pull request closed without merging**